File: Sprint02-Backend/src/main/java/com/br/saks/Sprint02/sprint02/model/Objetivo.java
package com.br.saks.Sprint02.sprint02.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Data;

//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
@Data
@Entity
public class Objetivo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = false, length = 200)
    private String nome;
    
    @Column(nullable = false)
    private int tempoConclusao;
    
    @Column(nullable = false, unique = false, length = 30)
    private float valorEstimado;
    
    @Column(nullable = false, unique = false, length = 20)
    private float valorEntrada;
    
    @Column(nullable = true, unique = false, length = 20)
    private String valorMensal;
    
    @ManyToOne(fetch=FetchType.EAGER)
    @JoinColumn(name = "id_investimento")
    private Investimento investimento;
    
    @ManyToOne
    @JoinColumn(name = "id_usuario", nullable = true)
    private Usuario usuario;
}

File: Sprint02-Frontend/lib/entities/objetivo.dart
import 'package:biblioteca_flutter/entities/usuario.dart';
import 'package:biblioteca_flutter/entities/investimento.dart';

import 'investimento.dart';

class Objetivo {
  int? id;
  String? nome;
  int? tempoConclusao;
  double? valorEstimado;
  double? valorEntrada;
  String? valorMensal;
  Usuario? usuario;
  Investimento? investimento;

  Objetivo(
      {this.id,
      this.nome,
      this.tempoConclusao,
      this.valorEntrada,
      this.valorEstimado,
      this.valorMensal,
      this.usuario,
      this.investimento});

  factory Objetivo.fromJson(Map<String, dynamic> json) {
    return Objetivo(
      id: json['id'],
      nome: json['nome'],
      tempoConclusao: json['tempoConclusao'],
      valorEntrada: json['valorEntrada'],
      valorEstimado: json['valorEstimado'],
      valorMensal: json['valorMensal'],
      usuario: Usuario.fromJson(json['usuario']),
      investimento: Investimento.fromJson(json['investimento']),
    );
  }
}
